packages/core/src/renderer/Callout.ts

```diff
@@ -0,0 +1,405 @@
+import { toScreen } from "utils/Util";
+import { attrFill, attrStroke, attrTitle } from "./AttrHelper";
+import { ShapeProps } from "./Renderer";
+
+import { angleBetween2, normalize } from "@thi.ng/vectors";
+import { zipWith, reduce, sortBy } from "lodash";
+
+// TODO: Move some of the below functions to a numeric utils for frontend?
+
+/**
+ * Some vector operations that can be used on lists.
+ */
+export const ops = {
```

Comment:
I guess these functions are here because there's no way to reuse the autodiff ops (e.g. adding polymorphic behavior to those functions are too costly)? In general,a separate module would be good for these functions. I imagine other complex shapes will need these operations too.

Reply:
Yeah, there's no way to make such ops nicely polymorphic because they have to call different functions depending on AD or not (e.g. AD vector `normalize` calls `add` whereas normal vector `normalize` calls the native JS `+`). I tried doing it all in AD but ran into #561, so had to do it in pure JS. I would be happy to move these into the utils module in the frontend.

Reply:
> I would be happy to move these into the utils module in the frontend.

Yeah that would be great!